pilot/pkg/controllers/ipallocate/ipallocate_test.go

```diff
@@ -477,6 +477,12 @@ func TestIPAllocateWithEnvCIDR(t *testing.T) {
 	features.IPAutoallocateIPv6Prefix = TestNonDefaultIPV6PrefixCIDR
 
 	rig := setupIPAllocateTest(t, TestNonDefaultIPV4Prefix, TestNonDefaultIPV6Prefix)
+
+	assert.EventuallyEqual(t, func() int {
+		se := rig.se.Get("pre-existing", "default")
+		return len(autoallocate.GetAddressesFromServiceEntry(se))
+	}, 2, retry.Converge(10), retry.Delay(time.Millisecond*5))
```

Comment:
Can we use `retry.MaxAttempts` to be consistent with the rest of the file?

```suggestion
	assert.EventuallyEqual(t, func() int {
		se := rig.se.Get("pre-existing", "default")
		return len(autoallocate.GetAddressesFromServiceEntry(se))
	}, 2, retry.MaxAttempts(10), retry.Delay(time.Millisecond*5))
```

Reply:
I tried using `retry.MaxAttempts(10)`, but it still fails occasionally—likely because it can give up early during controller startup. Switching to `retry.Converge(10)` makes it more stable since it waits for 10 consistent checks, helping avoid flaky tests.